### PR TITLE
Closing communication loop between minion and server.

### PR DESCRIPTION
**Pull request by splatch**
1. Cleaned up dependencies to horizon features.
2. Removed spring dependencies from snmp plugin.
3. Sink module for workflow result serialization.
4. Bridge WorkflowExecutionResult with MinionClient.

Workflow - MinionClient integration is done through `Consumer` call which is not very clever, but does the work.

Signed-off-by: Łukasz Dywicki <luke@code-house.org>

### CODE REVIEW

File: ipc/api/src/main/java/org/opennms/horizon/ipc/sink/aggregation/Aggregator.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+ *
+ * This class is designed to delegate dispatching to the calling threads as much
+ * as possible (those which make calls to {@link #aggregate(Message).}
+ *
+ * @author jwhite
+ *
+ * @param <S> individual message
+ * @param <T> aggregated message (i.e. bucket)
+ */
+public class Aggregator<S, T> implements AutoCloseable, Runnable {
```
Comment (mfrazier63) [inline ipc/api/src/main/java/org/opennms/horizon/ipc/sink/aggregation/Aggregator.java:57]: I smell a camel route brewing! :)

**Review by mfrazier63 — APPROVED** [review]
LGTM.

File: features/src/main/resources/features.xml [hunk 2]
```diff
@@ -151,13 +175,27 @@
     </feature>
 
     <feature name="snmp-plugins" description="OpenNMS SNMP Plugins" version="${project.version}">
+        <feature>plugins-api</feature>
```
[the remark this answers is not in the document]
Comment (splatch) [reply features/src/main/resources/features.xml:178]: No, not really, I was sorting out some of horizon deps and come to the point where I had to extract plugins api to test plugins alone can be activated. With above you can make `*-plugins` feature part of build time feature verification.

File: ipc/api/src/main/java/org/opennms/horizon/ipc/sink/aggregation/Aggregator.java (hunk 1 above)
Comment (splatch) [reply ipc/api/src/main/java/org/opennms/horizon/ipc/sink/aggregation/Aggregator.java:57]: Ahoy camel riders!

**Pull request merged**